sdk/storage/src/datalake/datalake_utilities.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "datalake/datalake_utilities.hpp"
+
+#include "common/crypt.hpp"
+#include "datalake/protocol/datalake_rest_client.hpp"
+
+#include <regex>
+
+namespace Azure { namespace Storage { namespace DataLake {
+  namespace Details {
+    UrlBuilder GetBlobUriFromDfsUri(const UrlBuilder& dfsUri)
```

Comment:
I'm not sure if this is going to work all the time. Because i've ever met blob endpoint like `account.zonenum.blob.windows.azure.net`, so I presume we also may have `account.zonenum.dfs.windows.azure.net`. Then your implementation of `GetBlobUriFromDfsUri` won't work.

Reply:
It's either what you said won't work or those customer domains that contains ".dfs." but does not mean dfs endpoint won't work. I think the latter has smaller chance so I'll remove the if statement, then we are aligned with the (buggy) .NET SDK's approach.